File: official/docs/csharp/current/luma/promise.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using EasyPost;
using Newtonsoft.Json;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            var client = new EasyPost.Client(new EasyPost.ClientConfiguration("EASYPOST_API_KEY"));

            EasyPost.Parameters.Luma.GetPromise parameters = new()
            {
                ToAddress = new EasyPost.Parameters.Address.Create
                {
                    Name = "Dr. Steve Brule",
                    Street1 = "5744 Silverton Ave",
                    City = "McKinney",
                    State = "TX",
                    Zip = "75070",
                    Country = "US",
                    Phone = "8573875756",
                    Email = "dr_steve_brule@gmail.com"
                },
                FromAddress = new EasyPost.Parameters.Address.Create
                {
                    Name = "EasyPost",
                    Street1 = "417 Montgomery Street",
                    Street2 = "5th Floor",
                    City = "San Francisco",
                    State = "CA",
                    Zip = "94104",
                    Country = "US",
                    Phone = "4153334445",
                    Email = "support@easypost.com"
                },
                Parcel = new EasyPost.Parameters.Parcel.Create
                {
                    Length = 20.2,
                    Width = 10.9,
                    Height = 5,
                    Weight = 65.9
                },
                RulesetName = "ruleset_...",
                PlannedShipDate = "2025-07-18",
                DeliverByDate = "2025-07-20"
            };

            EasyPost.Models.API.LumaInfo lumaInfo = await client.Luma.GetPromise(parameters);

            Console.WriteLine(JsonConvert.SerializeObject(lumaInfo, Formatting.Indented));
        }
    }
}

File: official/docs/golang/current/luma/promise.go
package example

import (
	"fmt"

	"github.com/EasyPost/easypost-go/v5"
)

func getPromise() {
	client := easypost.New("EASYPOST_API_KEY")

	lumaRequest := &easypost.LumaRequest{
		Shipment: easypost.Shipment{
			Parcel: &easypost.Parcel{
				Length: 20.2,
				Width:  10.9,
				Height: 5,
				Weight: 65.9,
			},
			ToAddress: &easypost.Address{
				Name:    "Dr. Steve Brule",
				Street1: "179 N Harbor Dr",
				City:    "Redondo Beach",
				State:   "CA",
				Zip:     "90277",
				Country: "US",
				Phone:   "4155559999",
				Email:   "dr_steve_brule@gmail.com",
			},
			FromAddress: &easypost.Address{
				Name:    "EasyPost",
				Street1: "417 Montgomery Street",
				Street2: "5th Floor",
				City:    "San Francisco",
				State:   "CA",
				Zip:     "90277",
				Country: "US",
				Phone:   "4155559999",
				Email:   "support@easypost.com",
			},
		},
		RulesetName:     "ruleset_...",
		PlannedShipDate: "2025-07-21",
		DeliverByDate:   "2025-07-25",
		PersistLabel:    false,
	}

	lumaInfo, _ := client.GetLumaPromise(lumaRequest)

	fmt.Println(lumaInfo)
}
